### PR TITLE
Make proxy destroy timeout configurable and increase it for the failing test

### PR DESCRIPTION
**Pull request by tkountis**
Fixes https://github.com/hazelcast/hazelcast-enterprise/issues/3445

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java [hunk 1]
```diff
@@ -347,6 +347,13 @@ private int getWhenNoSSLDetected() {
     public static final HazelcastProperty OPERATION_CALL_TIMEOUT_MILLIS
             = new HazelcastProperty("hazelcast.operation.call.timeout.millis", 60000, MILLISECONDS);
 
+    /**
+     * Timeout in seconds, used to wait for the proxy destroy operations
+     * (see. {@link com.hazelcast.spi.impl.proxyservice.ProxyService#destroyDistributedObject(String, String)} to finish.
+     */
+    public static final HazelcastProperty PROXY_DESTROY_TIMEOUT_SECONDS
```
Comment (devozerov) [inline hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java]: Do we need this property in the public API? The only place where it is required is `LocalIndexStatsTest`. If there are no demand for this from users, I would rather hide it inside a private API class. Or even set it through a setter method of a `ProxyServiceImpl` after the node started.

Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java]: Also, https://hazelcast.atlassian.net/wiki/spaces/EN/pages/253953578/Hazelcast+Configuration#HazelcastConfiguration-CritiqueofGroupProperties

Comment (vojtechtoman) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java]: I also thought the same but others were faster. To contribute something original to the debate, the javadoc of this property (if it stays in the public API) can be improved, namely the unclosed `(see...` part that does not parse for me.

Comment (tkountis) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java]: I listened to your demands, and I made it happen! its a private property now. I agree its very limited use-case so it doesn't even worth the private API, the private prop is a better fit.

File: hazelcast/src/test/java/com/hazelcast/map/LocalIndexStatsTest.java [hunk 2]
```diff
@@ -82,8 +84,10 @@ public void before() {
         mapName = randomMapName();
         noStatsMapName = mapName + "_no_stats";
 
+        System.setProperty("hazelcast.proxy.destroy.timeout.millis", Integer.toString(PROXY_DESTROY_TIMEOUT));
```
Comment (vojtechtoman) [inline hazelcast/src/test/java/com/hazelcast/map/LocalIndexStatsTest.java]: Clear the property once the test finishes?

**Comment by mmedenjak** [comment]
Failed with https://github.com/hazelcast/hazelcast/issues/16254, merging.

File: hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyServiceImpl.java [hunk 3]
```diff
@@ -102,6 +99,7 @@ public void handleException(Throwable throwable) {
 
     public ProxyServiceImpl(NodeEngineImpl nodeEngine) {
         this.nodeEngine = nodeEngine;
+        this.destroyTimeout = parseInt(System.getProperty("hazelcast.proxy.destroy.timeout.millis", "30"));
```
Comment (petrpleshachkov) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyServiceImpl.java]: Do we specify timeout in seconds or milliseconds? The name suggests milliseconds, but we interpret it in seconds?

**Pull request merged**